e2e-tests/development-runtime/plugins/gatsby-source-fake-data/gatsby-node.js

```diff
@@ -27,7 +27,9 @@ exports.sourceNodes = async function sourceNodes({
 
   if (webhookBody && webhookBody.items) {
     reporter.info(`Webhook data detected; creating nodes`)
-    webhookBody.items.forEach(node => createNode(api.getNode(node, helpers)))
+    webhookBody.items.forEach(node =>
+      createNode(api.addNode(api.getNode(node, helpers)))
+    )
```

Comment:
`api` wasn't tracking this, so resetting was never removing node create through explicit `webhookBody.items` way. This meant that this `gatsby-preview/updating.js` and `gatsby-preview/refreshing.js` only kind of worked when they were running in ... alphabetical order - if you would run them manually with `cy:open` in different order - those would fail (wat)

it also mean that to rerun those tests you would ... have to restart whole `gatsby develop` 😱 

I also need to mention here that fact that the node is not automatically cleaned up is weird - but this was like that since gatsby@2 (and likely earlier). So this need to be addressed - but it needs new APIs and is possibly breaking change - so I don't consider this a bug and rather a quirk ... (well it might be considered as bug - but it's how it works and changing it now would break whole lot of things than rely on this behaviour)